src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WrapWithTag/WrapWithTagEndpoint.cs

```diff
@@ -71,21 +71,23 @@ internal class WrapWithTagEndpoint : IWrapWithTagHandler
                 return documentSnapshot;
             }, cancellationToken).ConfigureAwait(false);
 
-            if (documentSnapshot is null || cancellationToken.IsCancellationRequested)
+            if (documentSnapshot is null)
             {
                 _logger.LogWarning($"Failed to find document {request.TextDocument.Uri}.");
                 return null;
             }
 
-            var codeDocument = await documentSnapshot.GetGeneratedOutputAsync();
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var codeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
             if (codeDocument.IsUnsupported())
             {
                 _logger.LogWarning($"Failed to retrieve generated output for document {request.TextDocument.Uri}.");
                 return null;
             }
 
             var sourceText = await documentSnapshot.GetTextAsync().ConfigureAwait(false);
-            if (!request.Range.Start.TryGetAbsoluteIndex(sourceText, _logger, out var hostDocumentIndex))
+            if (request.Range?.Start.TryGetAbsoluteIndex(sourceText, _logger, out var hostDocumentIndex) != true)
```

Comment:
Super nit:
```suggestion
            if (request.Range?.Start.TryGetAbsoluteIndex(sourceText, _logger, out var hostDocumentIndex) is false)
```

Reply:
Aside from the fact that I don't think like `is true` or `is false` (though can't really explain why), this changes semantics. The `?` here makes the result nullable, therefore we want to log if the result is false or null, or in other words if its not true. I guess `is not true` would work, but as mentioned, I don't like the `is` operator for that.

Reply:
It's a nit, so I'll just resolve since it's not your preference.